File: src/api/schemas/merchants.py
"""API schemas for merchant endpoints."""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class MerchantListItem(BaseModel):
    """Merchant summary in the directory list."""

    id: str
    name: str
    daily_volume: int = Field(..., alias="dailyVolume")
    risk_level: str = Field(..., alias="riskLevel")
    total_windows: int = Field(0, alias="totalWindows")
    spikes_detected: int = Field(0, alias="spikesDetected")
    fraud_count: int = Field(0, alias="fraudCount")
    organic_count: int = Field(0, alias="organicCount")
    review_count: int = Field(0, alias="reviewCount")

    model_config = {"populate_by_name": True}


class MerchantIncidentSummary(BaseModel):
    """Incident summary within a merchant profile."""

    id: str
    merchant_id: str = Field(..., alias="merchantId")
    date: str
    severity: str
    status: str
    classification: str
    fraud_probability: float = Field(..., alias="fraudProbability")
    confidence: float
    confidence_band: str = Field(..., alias="confidenceBand")
    anomaly_score: float = Field(..., alias="anomalyScore")
    decision_reason: str = Field("", alias="decisionReason")
    anomaly_summary: str = Field("", alias="anomalySummary")
    top_signals: list[str] = Field(default_factory=list, alias="topSignals")
    recommended_action: str = Field("", alias="recommendedAction")

    model_config = {"populate_by_name": True}


class MerchantProfileResponse(BaseModel):
    """Detailed merchant profile."""

    id: str
    name: str
    daily_volume: int = Field(..., alias="dailyVolume")
    risk_level: str = Field(..., alias="riskLevel")
    risk_posture: str = Field(..., alias="riskPosture")
    risk_label: str = Field(..., alias="riskLabel")
    summary: str
    total_windows: int = Field(0, alias="totalWindows")
    spikes_detected: int = Field(0, alias="spikesDetected")
    fraud_count: int = Field(0, alias="fraudCount")
    organic_count: int = Field(0, alias="organicCount")
    review_count: int = Field(0, alias="reviewCount")
    incidents: list[MerchantIncidentSummary] = Field(default_factory=list)

    model_config = {"populate_by_name": True}


class MerchantListResponse(BaseModel):
    """Response for GET /api/v1/merchants"""

    merchants: list[MerchantListItem]
    total: int

File: frontend/src/api/merchants.ts
/**
 * Merchant API client.
 *
 * Typed methods for calling the SUS backend merchant endpoints.
 */

import { apiClient, type ApiResponse } from './client';

// ---------------------------------------------------------------------------
// Backend response types
// ---------------------------------------------------------------------------

export interface BackendMerchantListItem {
  id: string;
  name: string;
  dailyVolume: number;
  riskLevel: 'low' | 'medium' | 'high';
  totalWindows: number;
  spikesDetected: number;
  fraudCount: number;
  organicCount: number;
  reviewCount: number;
}

export interface BackendMerchantListResponse {
  merchants: BackendMerchantListItem[];
  total: number;
}

export interface BackendMerchantIncident {
  id: string;
  merchantId: string;
  date: string;
  severity: string;
  status: string;
  classification: string;
  fraudProbability: number;
  confidence: number;
  confidenceBand: string;
  anomalyScore: number;
  decisionReason: string;
  anomalySummary: string;
  topSignals: string[];
  recommendedAction: string;
}

export interface BackendMerchantProfileResponse {
  id: string;
  name: string;
  dailyVolume: number;
  riskLevel: 'low' | 'medium' | 'high';
  riskPosture: 'normal' | 'watch' | 'high_attention';
  riskLabel: string;
  summary: string;
  totalWindows: number;
  spikesDetected: number;
  fraudCount: number;
  organicCount: number;
  reviewCount: number;
  incidents: BackendMerchantIncident[];
}

// ---------------------------------------------------------------------------
// API Methods
// ---------------------------------------------------------------------------

export async function getMerchants(
  params?: { search?: string; limit?: number },
): Promise<ApiResponse<BackendMerchantListResponse>> {
  const query = new URLSearchParams();
  if (params?.search) query.set('search', params.search);
  if (params?.limit) query.set('limit', String(params.limit));
  const qs = query.toString();
  return apiClient.get<BackendMerchantListResponse>(
    `/api/v1/merchants${qs ? '?' + qs : ''}`,
  );
}

export async function getMerchantProfile(
  merchantId: string,
): Promise<ApiResponse<BackendMerchantProfileResponse>> {
  return apiClient.get<BackendMerchantProfileResponse>(
    `/api/v1/merchants/${encodeURIComponent(merchantId)}`,
  );
}
